src/functions.js

```diff
@@ -0,0 +1,177 @@
+exports.exportConfluencePage = async function (event) {
+    const requestId = event.requestId;
+    if (!requestId) {
+        throw new Error('Request ID not provided');
+    }
+
+    const bucket = process.env.S3_BUCKET;
+    if (!bucket) {
+        throw new Error('Environment variable `S3_BUCKET` not provided');
+    }
+
+    const fileObj = tmp.fileSync({
+        postfix: '.ttl',
+        discardDescriptor: true
+    });
+
+    const writeStream = fs.createWriteStream(fileObj.name);
+    const exportConfig = _buildExportConfig({
+        ...event?.page,
+        output: writeStream
+    });
+
+    await wikiExport(exportConfig);
+    writeStream.close();
+
+    const hashStream = fs.createReadStream(fileObj.name);
+    const hash = await _hashFile('sha1', hashStream);
+    hashStream.close();
+
+    const s3 = new S3Client({});
+    const uploadStream = fs.createReadStream(fileObj.name);
+    const key = `${requestId}/${hash}.ttl`;
+
+    await s3.send(new PutObjectCommand({
+        Bucket: bucket,
+        Key: key,
+        Body: uploadStream
+    }));
+    uploadStream.close();
+
+    fileObj.removeCallback();
+
+    return {
+        path: `s3://${bucket}/${key}`,
+        sha1: hash
+    };
+}
+
+function _hashFile(algorithm, stream) {
+    return new Promise((resolve, reject) => stream
+        .on('error', reject)
+        .pipe(crypto.createHash(algorithm)
+            .setEncoding('hex'))
+        .once('finish', function () {
+            resolve(this.read())
+        })
+    );
+}
+
+exports.mapConfluencePageToChildren = async function (event) {
+    return _wikiChildPages(_buildExportConfig(event)).then(pages => pages.map(page => {
+        return {
+            page: page,
+            ...event.server && {server: event.server},
+            ...event.recurse && {recurse: event.recurse},
+            ...event.as_urls && {as_urls: event.as_urls}
+        };
+    }));
+}
+
+exports.popConfluencePageTree = async function (event) {
+    const stack = event.stack ?? [];
+    const limit = event.limit ?? 1;
+
+    const result = [];
+    for (let i = 0; i < limit; i++) {
+        const page = stack.pop();
+        if (!page) {
+            break;
+        }
+        result.push({
+            page: page,
+            ...event.server && {server: event.server},
+            ...event.as_urls && {as_urls: event.as_urls}
+        });
+        const children = await _wikiChildPages(_buildExportConfig({
+            ...event,
+            page: page,
+            recurse: false
+        }));
+        // push to stack in reverse order for left-to-right traversal
+        children.reduceRight((_, page) => stack.push(page), null);
+    }
+    return {
+        stack: stack,
+        limit: limit,
+        result: result,
+        ...event.server && {server: event.server},
+        ...event.as_urls && {as_urls: event.as_urls}
+    }
+}
+
+exports.generateUuid = async function (_event) {
```

Comment:
these functions don't need to be async. best practice is to only use async if `await` keyword is used.